distributed/worker.py

```diff
@@ -888,9 +893,10 @@ def __init__(
             lifetime_restart = dask.config.get("distributed.worker.lifetime.restart")
         self.lifetime_restart = lifetime_restart
 
-        if self.lifetime:
-            self.lifetime += (random.random() * 2 - 1) * lifetime_stagger
-            self.io_loop.call_later(self.lifetime, self.close_gracefully)
+        if lifetime:
+            lifetime += (random.random() * 2 - 1) * lifetime_stagger
+            self.io_loop.call_later(lifetime, self.close_gracefully)
+        self.lifetime = lifetime
```

Comment:
this was needed for mypy because previously `self.io_loop` was `Any` and `self.liftetime` is `Optional[float]` but `IOLoop.call_later` needs a `float`